tree_test.go

```diff
@@ -1385,13 +1456,17 @@ func TestManipulateChildren(t *testing.T) {
 }
 
 func TestLeafNodeInsert(t *testing.T) {
+	t.Parallel()
+
+	keyTest := make([]byte, len(ffx32KeyTest))
+	copy(keyTest, ffx32KeyTest)
 	values := make([][]byte, NodeWidth)
 	valIdx := 42
 	values[valIdx] = testValue
-	ln := NewLeafNode(ffx32KeyTest[:StemSize], values)
+	ln := NewLeafNode(keyTest[:StemSize], values)
 
 	// Check we get the value correctly via Get(...).
-	getValue, err := ln.Get(append(ffx32KeyTest[:StemSize], byte(valIdx)), nil)
+	getValue, err := ln.Get(append(keyTest[:StemSize], byte(valIdx)), nil)
```

Comment:
Here was the bug that became apparent when paralellizing (quite dangerous!).

The bug was in the deleted line: `append(ffx32KeyTest[:StemSize], byte(valIdx))`
That is literally appending bytes to `ffx32Keytest`, which is global. Now just make a copy in the test and do whatever we want.